engines/python/setup/djl_python/properties_manager/tnx_properties.py

```diff
@@ -0,0 +1,108 @@
+    level2 = 2
+    level3 = 3
+
+
+class Dtype(str, Enum):
+    f32 = 'fp32'
+    f16 = 'fp16'
+    bf16 = 'bf16'
+
+
+class TnXQuantizeMethods(str, Enum):
+    bitsandbytes8 = 'bitsandbytes8'
+
+
+TNX_SUPPORTED_ROLLING_BATCH_TYPES = ['auto']
+
+
+class TransformerNeuronXProperties(Properties):
+    """Transformer neuronx related configurations"""
+    neuron_optimize_level: Optional[OptimizeLevel] = None
+    dtype: Dtype = Dtype.f32
+    n_positions: int = 128
+    unroll: Optional[str] = None
+    load_in_8bit: Optional[bool] = False
+    low_cpu_mem_usage: bool = False
+    load_split_model: bool = False
+    context_length_estimate: Optional[dict] = None
+    amp: Optional[str] = None
+    quantize: Optional[TnXQuantizeMethods] = None
+
+    @field_validator('neuron_optimize_level')
+    def set_neuron_optimal_env(cls, level):
+        os.environ[
+            "NEURON_CC_FLAGS"] = os.environ["NEURON_CC_FLAGS"] + f" -O{level}"
+
+    @field_validator('context_length_estimate', mode='before')
+    def parse_context_length(cls, context_length_estimate):
+        return json.loads(context_length_estimate)
+
+    @field_validator('rolling_batch', mode='before')
+    def validate_rolling_batch(cls, rolling_batch: str) -> str:
+        if rolling_batch == RollingBatchEnum.disable.value:
+            return rolling_batch
+        if rolling_batch not in TNX_SUPPORTED_ROLLING_BATCH_TYPES:
+            logging.warning(
+                f"transformer neuronx only supports "
+                f"rolling batch type {TNX_SUPPORTED_ROLLING_BATCH_TYPES}."
+                f"choosing neuronx rolling batch automatically.")
+            return 'auto'
+        return rolling_batch
+
+    @field_validator('batch_size')
+    def validate_batch_size(cls, batch_size: int, fields) -> int:
+        """
+        Transformer neuronx has both option.batch_size and batch_size.
+        option.batch_size is to compile the model with batch size, which cannot be
+        differentiated in neuronx handlers. Hence, just throwing a warning here.
+        """
+        properties = fields.data
+        if batch_size > 1:
+            if properties[
+                    'rolling_batch'] == RollingBatchEnum.disable and properties[
+                        'enable_streaming'] != StreamingEnum.false:
+                logging.warning(
+                    "We cannot enable streaming for dynamic batching")
+        return batch_size
+
+    @model_validator(mode='after')
+    def set_amp_value(self, validation_info):
+        self.amp = self.dtype.name
+        return self
+
+    @model_validator(mode='after')
+    def set_quantize(self, validation_info):
+        if self.quantize and self.quantize.value == TnXQuantizeMethods.bitsandbytes8.value:
+            self.load_in_8bit = True
+        return self
+
+    @model_validator(mode='after')
+    def set_load_split_model(self, validation_info):
+        if self.load_split_model:
+            self.low_cpu_mem_usage = True
```

Comment:
@lanking520 Is this okay?

Reply:
Let's have @Lokiiiiii comments here. I think Split model loading doesn't need low_cpu_mem_usage

Reply:
`low_cpu_mem_usage` and `load_split_model` are not connected.

`low_cpu_mem_usage` is a known HuggingFace flag to save CPU memory by pipelining to Accelerator memory.
where as `load_split_model` was introduced to save CPU memory at the cost of DiskIO. 

I don't believe these orthogonal features should be bundled together.

Reply:
Thank you for commenting. Will make the changes.